### PR TITLE
Sinon: replace Bluebird with PromiseLike object

### PR DESCRIPTION
**Pull request by sameoldmadness**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

### CODE REVIEW

**Comment by sameoldmadness** [comment]
@43081j please take a look.

Fixes [this](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/33344#discussion_r259636796).

File: types/sinon/ts3.1/sinon-tests.ts [hunk 1]
```diff
@@ -424,10 +423,11 @@ function testSpy() {
 }
 
 function testStub() {
+    const promiseLike = { then: () => promiseLike };
```
Comment (SimonSchick) [inline types/sinon/ts3.1/sinon-tests.ts]: Eeeh, why not `const promiseLIke: PromiseLike = Promise.resolve()`
This seems like a weird construct.

**Review by SimonSchick — APPROVED** [review]

File: types/sinon/ts3.1/sinon-tests.ts (hunk 1 above)
Comment (43081j) [reply types/sinon/ts3.1/sinon-tests.ts]: yup i would go with that too. i was leaning more towards adding a promise-like interface last time but you are right that we can just add a `Promise.resolve`.

Comment (sameoldmadness) [reply types/sinon/ts3.1/sinon-tests.ts]: agreed, looks cleaner this way

fixed, thanks

File: types/sinon/ts3.1/sinon-tests.ts [hunk 2]
```diff
@@ -427,7 +426,7 @@ function testStub() {
     const obj = class {
         foo() { }
         promiseFunc() { return Promise.resolve('foo'); }
-        promiseLikeFunc() { return {} as any as Bluebird<string>; }
+        promiseLikeFunc() { return Promise.resolve('foo') as PromiseLike<string>; }
```
Comment (43081j) [inline types/sinon/ts3.1/sinon-tests.ts]: i would suggest putting it in a const to make this very clear and avoiding casting it.

```ts
const p: PromiseLike<string> = Promise.resolve('foo');
return p;
```

Comment (sameoldmadness) [reply types/sinon/ts3.1/sinon-tests.ts]: Could you explain the difference between casting and declaration with explicit type in this case?

As far as I understand, type checks are the same, e.g. one cannot cast an object that does not implement the interface.

Comment (43081j) [reply types/sinon/ts3.1/sinon-tests.ts]: There's likely very little difference, if any (I would have to do some investigation to be sure). it was more to make it clearer, but now that I read it again, it seems fine actually as we'd just be introducing bloat.

**Review by 43081j — APPROVED** [review]

**Pull request merged**